src/com/codefortomorrow/advanced/chapter16/solutions/LinkedList.java

```diff
@@ -0,0 +1,153 @@
+package com.codefortomorrow.advanced.chapter16.solutions;
+
+// UUID to represent each node's unique ID
+import java.util.UUID;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+// using Lombok for Getters and Setters (saves lines)
+import lombok.Setter;
+import lombok.Setter;
+
+/**
+ * @author ArmeetJatyani
+ * March 2021
+ *
+ * Implement a simple LinkedList
+ * You will need to create a LinkedListNode class, which represents each item/node in the linked list
+ * Required functionality...
+ * - head(): get head of list
+ * - tail(): get tail of list
+ * - add(): add to tail of list
+ * - push(): push to head of list
+ * - pop(): remove head of list
+ * - toString(): return a String representation of the list
+ */
+
+@Setter
+public class LinkedList {
+
+    private LinkedListNode head;
+
+    /**
+     * default constructor
+     */
+    public LinkedList() {
+        head = null;
+    }
+
+    /**
+     * constructor with first value
+     * @param value
+     */
+    public LinkedList(int value) {
+        // create first node
+        head = new LinkedListNode(value, null);
+    }
+
+    /**
+     * get head of Linked List
+     * @return
+     */
+    public LinkedListNode head() {
+        return this.head;
+    }
+
+    /**
+     * traverse and get tail of linked list
+     * @return
+     */
+    public LinkedListNode tail() {
+        LinkedListNode current = head;
+        if (current == null) return null;
+
+        while (current.getNext() != null) {
+            current = current.getNext();
+        }
+
+        return current;
+    }
+
+    /**
+     * add new element (node) to linked list
+     * @param value
+     */
+    public void add(int value) {
+        LinkedListNode tail = tail();
+        if (tail == null) {
+            head = new LinkedListNode(value, null);
+            return;
+        }
+
+        tail.setNext(new LinkedListNode(value, null));
+    }
+
+    /**
+     * push (add to head of linkedlist)
+     * @return
+     */
+    public void push(int value) {
+        LinkedListNode newHead = new LinkedListNode(value, head);
+        head = newHead;
+    }
+
+    /**
+     * pop (remove head of linkedlist)
+     * @return
+     */
+    public LinkedListNode pop() {
+        LinkedListNode popped = head;
+        head = head.getNext();
```

Comment:
```suggestion
        head = head.next();
```
Same reason as before; figured we might as well use the `next()` method you wrote